### PR TITLE
v0.19.0

### LINKED ISSUE #6: Need an official release
Currently these two lines 
https://github.com/conda-forge/libcugraph-feedstock/blob/ae9ef496797bee583b807d56cf52ba2dfbf0c3fd/recipe/conda_build_config.yaml#L45-L46
make the package published under the `libcugraph_dev` label. This is for development/release candidates. However, I think v0.19.0 is just released, and we should have a separate branch for the official releases, otherwise we cannot do
```
conda install -c conda-forge libcugraph
```

@rlratzel Could you do two things for me please?
1. Rename the current master to, say, `rc` so that it continues to publish packages under `libcugraph_dev `
2. Create a new branch (master) *without* the `libcugraph_dev` label that publishes v0.19.0 and the future releases 

This is blocking CuPy v9.0.0 (https://github.com/conda-forge/cupy-feedstock/pull/123).

### PR DESCRIPTION
**Pull request by leofang**
Close #6. ~~Merge this ***AFTER*** cloning the master branch, as this PR changes to the "main" label.~~ 
(Update: we don't need a rc branch, see https://github.com/conda-forge/libcugraph-feedstock/issues/6#issuecomment-824955999)

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by leofang** [comment]
@kkraus14 @rlratzel Any idea why raft's license file is not found?
> ValueError: License file given in about/license_file (/home/conda/recipe_root/cugraph/cpp/build/raft/src/raft/LICENSE) does not exist in source root dir or in recipe root dir (with meta.yaml)

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=307371&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=1081

**Comment by rlratzel** [comment]
> > ValueError: License file given in about/license_file (/home/conda/recipe_root/cugraph/cpp/build/raft/src/raft/LICENSE) does not exist in source root dir or in recipe root dir (with meta.yaml)

We may have changed the way raft is downloaded and built in our CMakeLists.txt file since the dev build, possibly resulting in the different directory (still confirming), so in the `meta.yaml` `license_file` list you could try changing this:
```
cugraph/cpp/build/raft/src/raft/LICENSE
```
to this:
```
cugraph/cpp/build/_deps/raft-src/LICENSE
```

**Comment by leofang** [comment]
Thanks for checking it out @rlratzel! One question: May I list myself as a maintainer? It's easier for me to have full control over CuPy's (optional) deps.

**Comment by rlratzel** [comment]
> Thanks for checking it out @rlratzel! One question: May I list myself as a maintainer? It's easier for me to have full control over CuPy's (optional) deps.

@leofang you're very welcome.  Yes, I think it's definitely a good idea to add yourself as a maintainer.  Let me know if you ever need any help.

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ about:
     - cugraph/LICENSE
     - rmm/LICENSE
     - cugraph/cpp/build/gunrock/src/gunrock_ext/LICENSE.TXT
-    - cugraph/cpp/build/raft/src/raft/LICENSE
+    - cugraph/cpp/build/_deps/raft-src/LICENSE
     - cugraph/cpp/build/_deps/thrust-src/LICENSE
     - cugraph/cpp/build/_deps/cuco-src/LICENSE
     - cugraph/cpp/build/cuhornet/src/cuhornet/LICENSE
```
Comment (leofang) [inline recipe/meta.yaml:57]: Looks like we need to fix this line too...
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=307538&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&t=b2a8456a-fb11-5506-ca32-5ccd32538dc0&l=1032
```suggestion
    - cugraph/cpp/build/_deps/cuhornet-src/LICENSE
```

Comment (rlratzel) [reply recipe/meta.yaml:57]: Yes, sorry about that. I was just typing that up here after I saw the failure.  I think that's the last one :)

Comment (leofang) [reply recipe/meta.yaml]: No worries, Rick! There're so many deps and it's hard to get them right in one shot...Let's see how it goes!

**Comment by leofang** [comment]
By the way, just curious, it took about 1 hr and a little more to build. Is this expected? Note the AWS CI limits the number of threads to 2 (or 1), so perhaps it is the reason?

**Comment by rlratzel** [comment]
> By the way, just curious, it took about 1 hr and a little more to build. Is this expected? Note the AWS CI limits the number of threads to 2 (or 1), so perhaps it is the reason?

Unfortunately that's not too surprising, especially if the amount of parallelism is that low.

**Comment by leofang** [comment]
@rlratzel CI is green 🎉

**Comment by leofang** [comment]
@rlratzel Could you check the CI for me? The merge commit is failing again, but this time when fetching cuhornet:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=307696&view=results
It seems something was changed in the past few hours?

**Comment by rlratzel** [comment]
> @rlratzel Could you check the CI for me? The merge commit is failing again, but this time when fetching cuhornet:
> https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=307696&view=results
> It seems something was changed in the past few hours?

It looks like there was a [commit to cuHornet](https://github.com/rapidsai/cuhornet/commits/main) a few hours ago, which caused the latest hash to change.  We have to use `FetchContent` in the `CMakeLists.txt` file with a hash instead of a tag since cuHornet currently doesn't define any tags (this is tech debt that needs to be addressed).  Since the hash is [hardcoded in our `CMakeLists.txt` file](https://github.com/rapidsai/cugraph/blob/9c0f7d1c9910bb3afa3615a33be75b3757ecb0ce/cpp/CMakeLists.txt#L279), the only thing I can think of would be to use `sed` in the recipe's build script to update it.  This seems like it will be a problem for any 0.19 build though, so I'm going to look into this further with the other cuGraph team members...

**Comment by leofang** [comment]
Sorry & Thanks @rlratzel  I didn't see your reply until now 🙏 I came up with a patch that seems to work, see #9.

**Comment by rlratzel** [comment]
> Sorry & Thanks @rlratzel I didn't see your reply until now 🙏 I came up with a patch that seems to work, see #9.

Thank you, yes that's perfect. We were just discussing removing `GIT_SHALLOW` as well as the correct fix for this.

**Comment by rlratzel** [comment]
@leofang - FYI, we just updated our [tagged v0.19.0 sources](https://github.com/rapidsai/cugraph/tree/v0.19.0) now with the fix (as mentioned [here](https://github.com/conda-forge/libcugraph-feedstock/pull/9#issuecomment-825280826)).

**Pull request merged**